store/sqlstore/post_store.go

```diff
@@ -1780,3 +1780,26 @@ func (s *SqlPostStore) SearchPostsInTeamForUser(paramsList []*model.SearchParams
 
 	return model.MakePostSearchResults(posts, nil), nil
 }
+
+func (s *SqlPostStore) GetOldestEntityCreationTime() (int64, *model.AppError) {
+	query := s.getQueryBuilder().Select("MIN(min_createat) min_createat").
+		Suffix(`FROM (
+					(SELECT MIN(createat) min_createat FROM Posts)
+					UNION
+					(SELECT MIN(createat) min_createat FROM Users)
+					UNION
+					(SELECT MIN(createat) min_createat FROM Channels)
+				) entities`)
```

Comment:
I think we can only look at user table. Rest will eventually be later than users table.. After the local mode (which there will be no users in the beginning so a team can be older than a user), we can add teams. It would be much smaller table to traverse.

Reply:
Yes, that is what I thought but asking Miguel he told me that some first users could be removed and that would lead to channels or posts to be the oldest one

Reply:
I agree this is the guaranteed way to get oldest entity, but I wonder if it would have a performance impact. Though the data in the tables might be sorted but still we can check if this approach is feasible.

Reply:
Impact in performance should be minimal because the createat fields are indexed and min has a low-performance footprint when doing queries.

Right now we're using a query that uses order by and limit in the Posts table and more impact than that is not going to have (users and channels tables are way smaller that posts table)